dap-ui.el

```diff
@@ -1109,5 +1109,27 @@ DEBUG-SESSION is the debug session triggering the event."
   (add-hook 'dap-breakpoints-changed-hook #'dap-ui-breakpoints--refresh)
   (add-hook 'kill-buffer-hook 'dap-ui-breakpoints--cleanup-hooks nil t))
 
+(defun dap-ui--window-visible-p (window-name)
+  "Return whether WINDOW-NAME is visible."
+  (-> (-compose 'buffer-name 'window-buffer)
+      (-map (window-list))
+      (-contains? window-name)))
+
+(defun dap-ui-show-debug-windows (_session)
+  "Show available debug windows for SESSION."
+  (save-excursion
+    (unless (dap-ui--window-visible-p dap-ui--locals-buffer)
```

Comment:
most likely we would need to have a list of windows to show up(or list of functions to call). We have also log window, loaded sources, expressions. The idea is to have a generic way the users to specify what windows to pop-up.